docker_cli/_exceptions.py

```diff
@@ -0,0 +1,125 @@
+from typing import Optional, overload
+
+
+class CommandNotFoundError(Exception):
+    """Raised when a command is attempted but not found."""
+
+    def __init__(self, command_name: str):
+        """
+        Raise this exception when a command is attempted but not found.
+
+        Parameters
+        ----------
+        command_name : str
+            The name of the command.
+        """
+        self.command_name = command_name
+        super(CommandNotFoundError, self).__init__(
+            f"Command '{self.command_name}' was run, but is not "
+            "present on the Docker image."
+        )
+
+
+class DockerBuildError(Exception):
+    """Raised when a docker image fails to build"""
+
+    @overload
+    def __init__(self, *, message: Optional[str]):
+        """
+        Raise this exception when a docker image fails to build.
+
+        Parameters
+        ----------
+        message : str
+            The error message.
+        """
+        ...
+
+    @overload
+    def __init__(self, *, image_tag: Optional[str], dockerfile: Optional[str]):
+        """
+        Raise this exception when a docker image fails to build.
+
+        Parameters
+        ----------
+        image_tag : str
+            The tag of the image.
+        dockerfile : str
+            The dockerfile used to build the image.
+        """
+        ...
+
+    def __init__(
+        self,
+        *,
+        message: Optional[str] = None,
+        image_tag: Optional[str] = None,
+        dockerfile: Optional[str] = None
```

Comment:
We shouldn't include type annotations here -- they should only be added to the `@overload` declarations above.

```suggestion
        message = None,
        image_tag = None,
        dockerfile = None
```